Source/ProgressionSystemRuntime/Private/Components/PSSpotComponent.cpp

```diff
@@ -28,16 +27,78 @@ void UPSSpotComponent::OnInitialized_Implementation()
 {
 	UPSWorldSubsystem& WorldSubsystem = UPSWorldSubsystem::Get();
 	WorldSubsystem.OnCurrentActiveSaveRowChanged.AddUniqueDynamic(this, &ThisClass::OnCurrentActiveSaveRowChanged);
+	WorldSubsystem.OnCurrentScoreChanged.AddUniqueDynamic(this, &ThisClass::OnCurrentScoreChanged);
+	BIND_ON_GAME_STATE_CHANGED(this, ThisClass::OnGameStateChanged);
+	constexpr bool bApplySkin = false;
+	RefreshAmountOfUnlockedSkins(bApplySkin);
+
 	// Save reference of this component to the world subsystem
 	WorldSubsystem.RegisterSpotComponent(this);
 }
 
+// Once the save file is reset the spot component needs to reset skins
+void UPSSpotComponent::OnReset_Implementation()
```

Comment:
This PR includes many changes I already reviewed, making it difficult to manage. While subreviews are generally a great approach, this one mixes in unrelated code.  

Please rebase it properly based on [this PR](https://github.com/moinm3uw/ProgressionSystem/pull/60) to avoid repeats or merge all new code into [the existing one](https://github.com/moinm3uw/ProgressionSystem/pull/60), so I can review everything together.